nym-vpn-desktop/src-tauri/src/main.rs

```diff
@@ -116,7 +105,10 @@ async fn main() -> Result<()> {
     // Read the env variables in the provided file and export them all to the local environment.
     nym_config::defaults::setup_env(app_config.env_config_file.clone());
 
-    let app_state = AppState::try_from((&app_data, &app_config)).map_err(|e| {
+    info!("Creating k/v embedded db");
+    let db = Db::new()?;
+
+    let app_state = AppState::try_from((&db, &app_config)).map_err(|e| {
```

Comment:
Imho from the users perspective it would be pretty neat to still continue to the UI if there as a failure here, and display it. So that people who don't run from the terminal also get to see what went wrong

Reply:
@octol if app state can't be recreated from the config file and/or db store, that means it will be 100% nonfunctional and the issue causing this is really an important and severe one (disk access refused, hard IO error etc).

Anyway, discussing that specific point is out of the scope of this PR:
please open a ticket if you really think it's necessary.